app/models/concerns/user_omniauth_concern.rb

```diff
@@ -0,0 +1,34 @@
+module UserOmniauthConcern
+  extend ActiveSupport::Concern
+  include UserOmniauthFacebookConcern
+
+  included do
+    devise :omniauthable
+  end
+
+  module ClassMethods
+    # Find or create a user from omniauth data.
+    #
+    # @param [OmniAuth::AuthHash] auth The data with omniauth info.
+    # @return [User] The user found or created from omniauth data.
+    def from_omniauth(auth)
+      identity = User::Identity.find_by(provider: auth.provider, uid: auth.uid)
```

Comment:
...seems like find_or* cannot handle associations.

maybe call this `find_or_create_by_identity()`?